packages/react-router/index.tsx

```diff
@@ -999,3 +999,23 @@ function resolvePathname(toPathname: string, fromPathname: string): string {
 
   return segments.length > 1 ? joinPaths(segments) : "/";
 }
+
+///////////////////////////////////////////////////////////////////////////////
+// DANGER! PLEASE READ ME!
+// We provide these exports as an escape hatch in the event that you need any
+// routing data that we don't provide an explicit API for. With that said, we
+// want to cover your use case if we can, so if you feel the need to use these
+// we want to hear from you. Let us know what you're building and we'll do our
+// best to make sure we can support you!
+//
+// We consider these exports an implementation detail and do not guarantee
+// against any breaking changes, regardless of the semver release. Use with
+// extreme caution and only if you understand the consequences. Godspeed.
+///////////////////////////////////////////////////////////////////////////////
+
+/** @internal */
```

Comment:
In conjunction with the `stripInternal` setting, this tells TypeScript not to include the exports in the type declarations. This may be additional incentive not to use them!

Reply:
Isn't the UNSAFE_ enough? As the contexts are typed internally at least it could break our CI on upgrade which is a good thing (and we don't mind as it's not covered by semver, but at least we know something broke).
The alternative would be to use module augmentation and maintain our own types, we will need to maintain them manually on every upgrade.